torch/csrc/jit/runtime/static/impl.h

```diff
@@ -151,6 +163,25 @@ class TORCH_API StaticRuntime {
     DCHECK(i < module_->output_regs.size());
     return reg_[module_->output_regs[i]];
   }
+  friend class MemoryPlanner;
+};
+
+class MemoryPlanner {
+ public:
+  MemoryPlanner(StaticRuntime* runtime);
+
+  void allocate_memory();
+  void clean_up_memory(std::vector<IValue>& reg);
```

Comment:
we can cache a reference to `std::vector<IValue>& reg` in the constructor of MemoryPlanner such that we can make the signature `clean_up_memory()`.

It might also be cleaner to just name these `allocate()` and `deallocate()`